### ISSUE TITLE
When we should use duration discriminator ?

### ISSUE THREAD

**Issue by mrakotos** [p0p4k/vits2_pytorch#104]
Hello, when we should use duration discriminator at the first 30k steps from scratch or after training a usable model then use duration discriminator for 30k. If we train with duration discriminator as the last steps (30k), we should apply noise to mas from the beginning of the training or use noise in mas with the duration discriminator (last 30k steps). Thank you.

**Comment by yukiarimo** [comment]
+1

**Comment by bzp83** [comment]
+1

**Comment by KiON-GiON** [comment]
In my case, using the adversarial duration discriminator added unstability to the voice. At least when used DD with SDP, the phoneme intervals were larger or sometimes the voice talked with an unnatural/drunk rhythm. Maybe it should be used at the last stages of the training, I don't know as I didn't read the VITS2 paper. Perhaps it's dataset-dependent. If your voice is dynamic (with varied rhythms, voice acting, etc. instead of a monotonous audiobook-style reading), using DD isn't recommended.

**Comment by yukiarimo** [comment]
If you have varied rhythms SDP is recommended. Just ensure your data is not trash!

**Comment by KiON-GiON** [comment]
@yukiarimo Yes, that's correct. I always use SDP, but can't say that the new duration discriminator adds a notable difference though. Have you used it and did you got usable results? (The "use_duration_discriminator" flag with type "dur_disc_2"). Now I use SDP alone without DD and it's more stable.
